pkg/storage/s3/s3.go

```diff
@@ -86,15 +91,31 @@ func NewImageStore(rootDir string, cacheDir string, gc bool, gcDelay time.Durati
 		linter:           linter,
 	}
 
-	cachePath := path.Join(cacheDir, CacheDBName+storage.DBExtensionName)
+	cachePath := path.Join(cacheDir, CacheDBName+constants.DBExtensionName)
 
 	if dedupe {
-		imgStore.cache = storage.NewCache(cacheDir, CacheDBName, false, log)
+		if len(cacheDriver) != 0 {
+			imgStore.cache = cacheDriver[0]
+		} else {
+			params := boltdb.DriverParameters{
+				RootDir:     cacheDir,
+				Name:        CacheDBName,
+				UseRelPaths: false,
+			}
+
+			imgStore.cache, _ = cachedbfactory.Create("boltdb", params, log)
+		}
 	} else {
 		// if dedupe was used in previous runs use it to serve blobs correctly
 		if _, err := os.Stat(cachePath); err == nil {
```

Comment:
I don't think this logic should be here? Can do this check when we validate the config, before initializing the controller and storage?